orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Execution.java

```diff
@@ -169,6 +169,11 @@ public void setKeepWaitingPipelines(boolean keepWaitingPipelines) {
    */
   @JsonIgnoreProperties(value = "execution")
   public @Nonnull List<Stage> getStages() {
+    if (stages.size() == 0) {
+      setStartTime(new Date().getTime());
```

Comment:
As per the @ezimanyi comment  on the  above issue, set the end time. "**buildTime"** is available in failure cases, but still not visible in UI on failure of pipeline.
The flow doesnot reach for success scenarios, it will reach the above flow only for failure scenarios. PFB screenshots for refernce:
![main](https://user-images.githubusercontent.com/48236703/63825669-48232780-c979-11e9-9d0f-6c14533a8f0e.png)

Reply:
Hi @ezimanyi , As per the logs, for startup failure of the pipeline, the flow goes directly to no stages block form there it is redirecting to getApplications which is showing as "Completed". There is no chance to set the endpoint after the getStages() method. Please find the below logs for reference :::
2019-08-29 07:23:47.840  INFO 20215 --- [.1-8083-exec-10] c.n.s.o.c.OperationsController           : [anonymous] Started pipeline 01DKE44HS0NZ4S1K7H3YNXCJ6H based on request body {"application":"neelimatest3830","name":"testfailpubsub","id":"2c2e3bb1-095a-44f6-bf7f-2bfd78c77143","executionId":"01DKE44HS0NZ4S1K7H3YNXCJ6H","parallel":false,"disabled":false,"limitConcurrent":true,"keepWaitingPipelines":false,"plan":false,"respectQuietPeriod":false,"triggers":[{"id":"2f5094f7-7116-3a2c-876d-4775886784cf","type":"pubsub","enabled":true,"rebake":false,"dryRun":false}],"stages":[],"notifications":[],"receivedArtifacts":[],"expectedArtifacts":[],"trigger":{"type":"manual","enabled":false,"executionId":"01DKE44HS0NZ4S1K7H3YNXCJ6H","user":"[anonymous]","rebake":false,"dryRun":false,"eventId":"b33e65fa-a984-4f9b-b002-8b01e46a9a66"}}
2019-08-29 07:23:47.851  INFO 20215 --- [    scheduler-1] com.netflix.spinnaker.q.QueueProcessor   : [] Received message StartExecution(executionType=pipeline, executionId=01DKE44HS0NZ4S1K7H3YNXCJ6H, application=neelimatest3830)
2019-08-29 07:23:47.870  WARN 20215 --- [    handlers-10] c.n.s.o.q.handler.StartExecutionHandler  : [] No initial stages found (executionId: 01DKE44HS0NZ4S1K7H3YNXCJ6H)
2019-08-29 07:23:47.886  INFO 20215 --- [      events-16] c.n.s.orca.front50.Front50Service        : [anonymous] ---> HTTP GET http://localhost:8080/notifications/application/neelimatest3830
2019-08-29 07:23:47.890  INFO 20215 --- [       events-8] c.n.s.orca.front50.Front50Service        : [] ---> HTTP GET http://localhost:8080/pipelines?restricted=false
2019-08-29 07:23:48.020  INFO 20215 --- [       events-8] c.n.s.orca.front50.Front50Service        : [] <--- HTTP 200 http://localhost:8080/pipelines?restricted=false (129ms)

Reply:
Hm, I'm not as familiar as others are with the flow for failed pipelines, but there needs to be a solution here that doesn't involve mutating an object in an accessor, even if it does involve some amount of refactoring of the control flow.